contrib/storage-splunk/src/main/java/org/apache/drill/exec/store/splunk/SplunkSchema.java

```diff
@@ -94,6 +109,10 @@ public CreateTableEntry createNewTable(String tableName, List<String> partitionC
         .message(plugin.getName() + " is not writable.")
         .build(logger);
     }
+    // Clear the index cache.
+    if (useCache) {
+      cache.invalidate(getNameForCache());
```

Comment:
It feels like it would be more natural (and efficient) for the cache to hold one entry per Splunk index, rather than a single entry containing the list of all indexes. Is there a reason it isn't built that way?

Reply:
@jnturton I think that's exactly what it does.  The `invalidate` is just the delete method, so the code there removes any cache entries with that entry.   Also as an FYI, the cache adds the username to the index name so that if user translation is enabled, users will not see other users' cache.